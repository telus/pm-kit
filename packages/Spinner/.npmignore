*.stories.{js,mdx}
rollup.config.js
Spinner.jsx
__tests__/