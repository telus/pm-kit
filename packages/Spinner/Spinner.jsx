import React from 'react'
import { css } from '@emotion/core'
import { parkGreen } from '@pm-kit/colours'
import PropTypes from 'prop-types'

const Spinner = ({ size }) => {
  const loader = css`
    @keyframes spinner {
      to {
        transform: rotate(360deg);
      }
    }
    border: 3px solid ${parkGreen};
    border-radius: 50%;
    border-right-color: transparent;
    display: inline-block;
    width: ${size};
    height: ${size};
    animation: 1s linear infinite spinner;
    animation-delay: var(--spinner-delay);
  `

  return <div css={loader}></div>
}

Spinner.propTypes = {
  /**
   * The height and width of spinner.
   */
  size: PropTypes.string,
}

Spinner.defaultProps = {
  size: '100%',
}

export default Spinner
