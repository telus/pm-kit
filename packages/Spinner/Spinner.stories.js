import React from 'react'
import Spinner from './Spinner'
import { css } from '@emotion/core'
import { withKnobs, select } from '@storybook/addon-knobs'

export default {
  title: 'Design System|Spinner',
  component: Spinner,
  decorators: [withKnobs],
}

export const Default = () => {
  const container = css`
    width: 44px;
    height: 44px;
  `
  return (
    <div css={container}>
      <Spinner />
    </div>
  )
}

export const Playground = () => {
  return <Spinner size={select('spinner', ['100%', '24px', '34px', '44px'], '100%')} />
}

Playground.story = {
  name: 'playground',
  parameters: { docs: { page: null, disable: true } },
}
