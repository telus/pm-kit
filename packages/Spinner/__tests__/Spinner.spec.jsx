import React from 'react'
import { render } from 'enzyme'

import Spinner from '../Spinner'

describe('Spinner', () => {
  it('renders', () => {
    const spinner = render(<Spinner />)

    expect(spinner).toMatchSnapshot()
  })
})
