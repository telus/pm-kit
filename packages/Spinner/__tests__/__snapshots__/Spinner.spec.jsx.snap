// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner renders 1`] = `
.emotion-0 {
  border: 3px solid #034045;
  border-radius: 50%;
  border-right-color: transparent;
  display: inline-block;
  width: 100%;
  height: 100%;
  -webkit-animation: 1s linear infinite spinner;
  animation: 1s linear infinite spinner;
  -webkit-animation-delay: var(--spinner-delay);
  animation-delay: var(--spinner-delay);
}

<div
  class="emotion-0"
/>
`;
